tiledb/sm/consolidator/group_meta_consolidator.cc

```diff
@@ -75,59 +75,33 @@ Status GroupMetaConsolidator::consolidate(
 
   // Open group for reading
   auto group_uri = URI(group_name);
-  Group group_for_reads(group_uri, storage_manager_);
-  RETURN_NOT_OK(group_for_reads.open(
-      QueryType::READ, config_.timestamp_start_, config_.timestamp_end_));
+  AutoCloseGroup group_for_reads(
+      group_uri,
+      storage_manager_,
+      QueryType::READ,
+      config_.timestamp_start_,
+      config_.timestamp_end_);
 
   // Open group for writing
-  Group group_for_writes(group_uri, storage_manager_);
-  RETURN_NOT_OK_ELSE(
-      group_for_writes.open(QueryType::WRITE),
-      throw_if_not_ok(group_for_reads.close()));
+  AutoCloseGroup group_for_writes(
+      group_uri, storage_manager_, QueryType::WRITE);
 
   // Swap the in-memory metadata between the two groups.
   // After that, the group for writes will store the (consolidated by
   // the way metadata loading works) metadata of the group for reads
-  Metadata* metadata_r;
-  auto st = group_for_reads.metadata(&metadata_r);
-  if (!st.ok()) {
-    throw_if_not_ok(group_for_reads.close());
-    throw_if_not_ok(group_for_writes.close());
-    return st;
-  }
-  Metadata* metadata_w;
-  st = group_for_writes.metadata(&metadata_w);
-  if (!st.ok()) {
-    throw_if_not_ok(group_for_reads.close());
-    throw_if_not_ok(group_for_writes.close());
-    return st;
-  }
+  Metadata* metadata_r{group_for_reads->metadata()};
+  Metadata* metadata_w{group_for_writes->metadata()};
   metadata_r->swap(metadata_w);
 
   // Metadata uris to delete
   const auto to_vacuum = metadata_w->loaded_metadata_uris();
 
   // Generate new name for consolidated metadata
-  st = metadata_w->generate_uri(group_uri);
-  if (!st.ok()) {
-    throw_if_not_ok(group_for_reads.close());
-    throw_if_not_ok(group_for_writes.close());
-    return st;
-  }
+  RETURN_NOT_OK(metadata_w->generate_uri(group_uri));
```

Comment:
We're not using `RETURN_NOT_OK` in new code. Use `throw_if_not_ok`.

If `throw_if_not_ok` doesn't work, there's something elsewhere that would need to change. I don't think we've had such a case yet, but it's worth keeping an eye out for it.